### PR TITLE
HADOOP-17288. Use shaded guava from thirdparty. (branch-3.3)

### PR DESCRIPTION
**Pull request by ayushtkn**
https://issues.apache.org/jira/browse/HADOOP-17288

### CODE REVIEW

**Comment by ayushtkn** [comment]
The test results are at 
https://ci-hadoop.apache.org/blue/organizations/jenkins/hadoop-multibranch/detail/PR-2505/1/tests

Not sure, why there is no comment from jenkins here

File: Jenkinsfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ pipeline {
 
     options {
         buildDiscarder(logRotator(numToKeepStr: '5'))
-        timeout (time: 20, unit: 'HOURS')
+        timeout (time: 35, unit: 'HOURS')
```
Comment (saintstack) [inline Jenkinsfile:26]: This because so much has been touched? We need the longer time to build?

**Review by saintstack — COMMENTED** [review]
Skimmed. LGTM. What did you have to change from master PR?

Failures don't look to be obviously related. Do you recognize any of these failures as flakies? Regular failings? I tried to retrigger the build but don't see how. Perhaps try a new push to see what failures we see in a re-run?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  30m  7s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m 11s |  https://github.com/apache/hadoop/pull/2505 does not apply to branch-3.3. Rebase required? Wrong Branch? See https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2505 |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/3/console |
| versions | git=2.7.4 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 49s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   1m 13s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  1s |  The patch does not contain any @author tags.  |
| -1 :x: |  pathlen  |   0m  0s |  The patch appears to contain 5 files with names longer than 240  |
| +1 :green_heart: |  test4tests  |   0m  1s |  The patch appears to include 413 new or modified test files.  |
||| _ branch-3.3 Compile Tests _ |
| +0 :ok: |  mvndep  |   4m  1s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  28m 14s |  branch-3.3 passed  |
| +1 :green_heart: |  compile  |  15m 40s |  branch-3.3 passed  |
| +1 :green_heart: |  checkstyle  |  27m 21s |  branch-3.3 passed  |
| +1 :green_heart: |  mvnsite  |  22m  7s |  branch-3.3 passed  |
| +1 :green_heart: |  shadedclient  |  15m  0s |  branch has no errors when building and testing our client artifacts.  |
| -1 :x: |  javadoc  |   5m 27s |  root in branch-3.3 failed.  |
| +0 :ok: |  spotbugs  |   0m 46s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +0 :ok: |  findbugs  |   0m 26s |  branch/hadoop-project no findbugs output file (findbugsXml.xml)  |
| -1 :x: |  findbugs  |  29m 46s |  root in branch-3.3 has 3 extant findbugs warnings.  |
| +0 :ok: |  findbugs  |   0m 32s |  branch/hadoop-client-modules/hadoop-client-minicluster no findbugs output file (findbugsXml.xml)  |
| -1 :x: |  findbugs  |   0m 44s |  hadoop-cloud-storage-project/hadoop-cos in branch-3.3 has 1 extant findbugs warnings.  |
| +0 :ok: |  findbugs  |   0m 36s |  branch/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests no findbugs output file (findbugsXml.xml)  |
| -1 :x: |  findbugs  |   0m 49s |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase-tests in branch-3.3 has 2 extant findbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 38s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  59m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 29s |  the patch passed  |
| +1 :green_heart: |  javac  |  15m 29s |  the patch passed  |
| -0 :warning: |  checkstyle  |  27m 20s |  root: The patch generated 191 new + 25239 unchanged - 13 fixed = 25430 total (was 25252)  |
| +1 :green_heart: |  mvnsite  |  20m 32s |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  shelldocs  |   0m 22s |  There were no new shelldocs issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 49s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedclient  |  15m 51s |  patch has no errors when building and testing our client artifacts.  |
| -1 :x: |  javadoc  |   5m 28s |  root in the patch failed.  |
| +0 :ok: |  findbugs  |   0m 25s |  hadoop-project has no data from findbugs  |
| +0 :ok: |  findbugs  |   0m 31s |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests has no data from findbugs  |
| +0 :ok: |  findbugs  |   0m 29s |  hadoop-client-modules/hadoop-client-minicluster has no data from findbugs  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 555m 38s |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 45s |  The patch does not generate ASF License warnings.  |
|  |   | 1029m 57s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.applications.distributedshell.TestDistributedShell |
|   | hadoop.yarn.server.resourcemanager.scheduler.fair.TestFairSchedulerPreemption |
|   | hadoop.yarn.server.resourcemanager.TestRMHATimelineCollectors |
|   | hadoop.tools.dynamometer.TestDynamometerInfra |
|   | hadoop.hdfs.server.namenode.ha.TestHAAppend |
|   | hadoop.hdfs.server.datanode.TestBPOfferService |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2505 |
| Optional Tests | dupname asflicense shellcheck shelldocs compile javac javadoc mvninstall mvnsite unit shadedclient xml findbugs checkstyle |
| uname | Linux 948f20ecfece 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / 4628647 |
| Default Java | Private Build-1.8.0_275-8u275-b01-0ubuntu1~16.04-b01 |
| pathlen | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/pathlen.txt |
| javadoc | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/branch-javadoc-root.txt |
| findbugs | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/branch-findbugs-root-warnings.html |
| findbugs | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/branch-findbugs-hadoop-cloud-storage-project_hadoop-cos-warnings.html |
| findbugs | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/branch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-timelineservice-hbase-tests-warnings.html |
| checkstyle | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/diff-checkstyle-root.txt |
| javadoc | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/patch-javadoc-root.txt |
| unit | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/artifact/out/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/testReport/ |
| Max. process+thread count | 4379 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-maven-plugins hadoop-common-project/hadoop-auth hadoop-common-project/hadoop-common hadoop-common-project/hadoop-nfs hadoop-common-project/hadoop-kms hadoop-common-project/hadoop-registry hadoop-hdfs-project/hadoop-hdfs-client hadoop-hdfs-project/hadoop-hdfs hadoop-hdfs-project/hadoop-hdfs-httpfs hadoop-hdfs-project/hadoop-hdfs-nfs hadoop-hdfs-project/hadoop-hdfs-rbf hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-web-proxy hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-applicationhistoryservice hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-tests hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-sharedcachemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timeline-pluginstorage hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase/hadoop-yarn-server-timelineservice-hbase-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase/hadoop-yarn-server-timelineservice-hbase-client hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase/hadoop-yarn-server-timelineservice-hbase-server/hadoop-yarn-server-timelineservice-hbase-server-1 hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase-tests hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-documentstore hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-applications-distributedshell hadoop-mapreduce-project/hadoop-mapreduce-client hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-common hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-shuffle hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-core hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-api hadoop-yarn-project/hadoop-yarn/hadoop-yarn-csi hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-nativetask hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-uploader hadoop-mapreduce-project/hadoop-mapreduce-examples hadoop-tools/hadoop-distcp hadoop-tools/hadoop-dynamometer/hadoop-dynamometer-workload hadoop-tools/hadoop-dynamometer/hadoop-dynamometer-infra hadoop-tools/hadoop-archives hadoop-tools/hadoop-archive-logs hadoop-tools/hadoop-gridmix hadoop-tools/hadoop-aws hadoop-tools/hadoop-kafka hadoop-tools/hadoop-azure hadoop-tools/hadoop-aliyun hadoop-tools/hadoop-sls hadoop-tools/hadoop-azure-datalake hadoop-tools/hadoop-fs2img hadoop-client-modules/hadoop-client-minicluster hadoop-cloud-storage-project/hadoop-cos . hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-hbase/hadoop-yarn-server-timelineservice-hbase-server/hadoop-yarn-server-timelineservice-hbase-server-2 U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/4/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.3.7 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: Jenkinsfile (hunk 1 above)
Comment (ayushtkn) [reply Jenkinsfile]: I increased it, Luckily got the result this time

**Comment by ayushtkn** [comment]
@saintstack  Thanx for the review.
To be precise, I didn't change anything in the pom.xml's, I just removed the java code from trunk and applied the patch to branch-3.3 and luckily it had no conflicts, Then I just compiled and the java code got regenerated wrt branch-3.3,

There is a javadoc build failure, that is there in branch-3.3 without my patch as well for aws,
Test failures doesn't look related

**Review by saintstack — APPROVED** [review]
+1 from me

(We have conflicts now?)

**Comment by ayushtkn** [comment]
Thanx @saintstack for the review, I have rebased and merged this to branch-3.3

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m 10s |  https://github.com/apache/hadoop/pull/2505 does not apply to branch-3.3. Rebase required? Wrong Branch? See https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/2505 |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2505/5/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**